src/packages/recompose/flattenProp.js

```diff
@@ -1,15 +1,11 @@
+import mapProps from './mapProps'
 import omit from './utils/omit'
 import createHelper from './createHelper'
-import createEagerFactory from './createEagerFactory'
 
-const flattenProp = propName => BaseComponent => {
-  const factory = createEagerFactory(BaseComponent)
-  return props => (
-    factory({
-      ...omit(props, [propName]),
-      ...props[propName]
-    })
-  )
-}
+const flattenProp = propName =>
+  mapProps(props => ({
+    ...omit(props, [propName]),
+    ...props[propName]
+  }))
```

Comment:
In case it's not clear why this is included in this PR (other than being the right way to define `flattenProp`, anyway), the idea is that there are some helpers that we don't want to define using middleware, because middleware necessitates using a class component, even when a functional component would suffice. What we really want is to use a functional component *unless* the HOC is composed with another HOC which *must* use a class. Put another way, a single functional component is better than a class component, but a single class component is better than a class component and a functional component.

To achieve this, we will special case `mapProps` to use a functional component by default, but use middleware when being composed with other middleware-using HOCs.